pkg/mcs/scheduling/server/server.go

```diff
@@ -506,7 +506,7 @@ func (s *Server) startCluster(context.Context) error {
 		return err
 	}
 	s.hbStreams = hbstream.NewHeartbeatStreams(s.Context(), constant.SchedulingServiceName, s.basicCluster)
-	cluster, err := NewCluster(s.Context(), s.persistConfig, s.storage, s.basicCluster, s.hbStreams, s.checkMembershipCh, s.GetHTTPClient(), s.GetBackendEndpoints())
+	cluster, err := NewCluster(s.Context(), s.persistConfig, s.storage, s.basicCluster, s.hbStreams, s.checkMembershipCh, s.GetHTTPClient(), s.GetBackendEndpoints(), s.participant)
```

Comment:
A quick thought: all parameters used by `NewCluster` come from the `*Server`. Can we optimize its function signature to avoid redundant parameters?